st2client/tests/unit/test_shell.py

```diff
@@ -617,6 +618,7 @@ def _write_mock_config(self):
         with open(self._mock_config_path, "w") as fp:
             fp.write(MOCK_CONFIG)
 
+    @pytest.mark.skipif(os.getuid() == 0, reason="Test must be run as non-root user.")
```

Comment:
The test is to confirm access to the cache file is denied when file permissions at set to `0o000`.  The test fails when run as root because root isn't denied access to the file so I added the `skipif`.  If there are failures in the CI and it's not running as root, then there may be other issues to resolve.  I'd say the fact the test is failing and not being skipped indicates GHA isn't running as root and there's a problem to be resolved.

Reply:
I cherry picked the test skips, but it's still falling, so at least GHA is not using root.

I wonder if I missed cherry picking something...

Reply:
It seems I missed cherry-picking ca7db1d1d9f1db9a25f91380475d19df2995f0e2. Oops. Plus I had to adjust a few more tests to use `warning` instead of `warn`. Now I've got this passing in #6188.